lib/puppet/provider/package/yum.rb

```diff
@@ -13,6 +13,30 @@
 
   commands :cmd => "yum", :rpm => "rpm"
 
+  YUM_ARCH_LIST = [
+    'i386',
+    'i686',
+    'ppc',
+    'ppc64',
+    'armv3l',
+    'armv4b',
+    'armv4l',
+    'armv4tl',
+    'armv5tel',
+    'armv5tejl',
+    'armv6l',
+    'armv7l',
+    'm68kmint',
+    's390',
+    's390x',
+    'ia64',
+    'x86_64',
+    'sh3',
+    'sh4',
+  ]
+
+  ARCH_REGEX = Regexp.new(YUM_ARCH_LIST.join('|\.'))
```

Comment:
Technically know, but if it's not you're doing it wrong according to the docs:

> While RPM will run just as well if a package file has been renamed, when the packages are created during RPM's build process, they follow a specific naming convention. The convention is:
name-version-release.architecture.rpm 
- http://www.rpm.org/max-rpm/ch-rpm-file-format.html

Do we want to cover the case of incorrectly named RPMs?